src/private/MDMPerformerGroup.m

```diff
@@ -93,28 +100,55 @@ - (void)terminateIsActiveToken:(id<MDMIsActiveTokenable>)token
 
 #pragma mark - Private
 
-- (id<MDMPerforming>)performerForPlan:(id<MDMPlan>)plan isNew:(BOOL *)isNew {
+- (void)removePlanNamed:(nonnull NSString *)name from:(nonnull id)target withPerformerInfo:(nullable MDMPerformerInfo *)performerInfo {
+  if (performerInfo) {
+    id<MDMPerforming> performer = performerInfo.performer;
+    if ([performer respondsToSelector:@selector(removePlanNamed:)]) {
+      [(id<MDMNamedPlanPerforming>)performer removePlanNamed:name];
+    }
+  } else {
+    // this is the case whereby the client has tried to remove a named performer which was never added in the first place
+  }
```

Comment:
We can simplify this implementation by leaning on nil-messaging behavior.

- [ ] Remove the if/else statement.